LFPy/cell.py

```diff
@@ -1917,5 +1913,332 @@ def insert_v_ext(self, v_ext, t_ext):
+        d_list : ndarray [microm]
+            Array of distance vectors traveled by each axial
+            current in i_axial.
+        i_axial : ndarray [nA]
+            Array of axial currents, I(t) going from compartment
+            end to mid/ compartment mid to start for all
+            compartment halves in cell.
+        """
+        if not hasattr(self, 'vmem'):
+            raise AttributeError('no vmem, run cell.simulate(rec_vmem=True)')
+
+        iaxial = []
+        d_list = []
+
+        # dseg = np.array(zip((self.xmid - self.xstart),
+        #            (self.ymid - self.ystart),
+        #            (self.zmid - self.zstart)))
+        # dpar = np.array(zip((self.xend - self.xmid),
+        #            (self.yend - self.ymid),
+        #            (self.zend - self.zmid)))
+
+        dseg = np.c_[self.xmid - self.xstart,
+                     self.ymid - self.ystart,
+                     self.zmid - self.zstart]
+        dpar = np.c_[self.xend - self.xmid,
+                     self.yend - self.ymid,
+                     self.zend - self.zmid]
+
+        children_dict = self.get_dict_of_children_idx()
+        for sec in neuron.h.allsec():
+            if not neuron.h.SectionRef(sec.name()).has_parent():
+                # skip soma, since soma is an orphan
+                continue
+
+            bottom_seg = True
+
+            parentseg = neuron.h.SectionRef(sec.name()).parent()
+            parentsec = parentseg.sec
+
+            branch = len(children_dict[parentsec.name()]) > 1
+
+            parent_idx = self.get_idx(section=parentsec.name())[-1]
+            seg_idx = self.get_idx(section=sec.name())[0]
+
+            for _ in sec:
+                iseg, ipar = self._parent_and_segment_i(seg_idx, parent_idx,
+                                                        bottom_seg,branch,
+                                                        parentsec)
+
+                if bottom_seg and 'soma' in parentsec.name():
+                    # if a seg is connented to soma, it is
+                    # connected to the middle of soma,
+                    # and dpar needs to be altered.
+                    dpar[parent_idx] = np.array([self.xstart[seg_idx] -
+                                         self.xmid[parent_idx],
+                                        self.ystart[seg_idx] -
+                                         self.ymid[parent_idx],
+                                        self.zstart[seg_idx] -
+                                         self.zmid[parent_idx]])
+
+                d_list.append(dpar[parent_idx])
+                d_list.append(dseg[seg_idx])
+                iaxial.append(ipar)
+                iaxial.append(iseg)
+
+                parent_idx = seg_idx
+                seg_idx += 1
+                branch = False
+                bottom_seg = False
+                parent_ri = 0
+
+        return np.array(d_list), np.array(iaxial)
+
+
+    def get_axial_resistance(self):
```

Comment:
Also here the description is a bit imprecise, it is the axial resistance from the parent segment, not the midpoint of the parent segment, see the description of `neuron.h.ri`: 

>     Return the resistance (in megohms) between the center of the segment containing x and its parent segment. This can be used to compute axial current given the voltage at two adjacent points. If there is no parent the "infinite" resistance returned is 1e30.